examples/asset/extra_source.rs

```diff
@@ -0,0 +1,51 @@
+//! An example of registering an extra asset source, and loading assets from it.
+//! This asset source exists in addition to the default asset source.
+
+use bevy::asset::{
+    io::{AssetSourceBuilder, AssetSourceBuilders, AssetSourceId},
+    AssetPath,
+};
+use bevy::prelude::*;
+use std::path::Path;
+
+fn main() {
+    let mut app = App::new();
+
+    // We add an extra asset source with the name "example_files" to the
+    // AssetSourceBuilders.
+    // This needs to be done before AssetPlugin finalizes building them
+    let mut sources = app
```

Comment:
Is there a reason why you're not using `.register_asset_source()`?

```rust
.register_asset_source(
    "example_files",
    /// etc.
)
```

Reply:
Don't think so, I might've missed that method, since the bevy_asset docs are fairly sparse and don't mention this method on relevant types. I made this on 0.12 so we didn't have the cool examples in docs thing yet either

Reply:
Understood. I think that using `.register_asset_source()` would make this example better, as it doesn't require the user to read-modify-write part of the app state. I've been using it in my own apps and it works nicely. I think this what we want to encourage users to do.

Just as an aside: so far, the biggest use case (for me) for asset sources is library crates with their own assets.